lib/IR/IRGen.cpp

```diff
@@ -245,12 +245,27 @@ struct IRGenVisitor : NodeWalker {
       std::vector<size_t> offsets(CC->dims().size(), 0);
       unsigned dim = CC->getDim();
 
-      for (int i = 0, e = inputs.size(); i < e; i++) {
-        builder_.createInsertTensorInst(CC->getName(), dest,
-                                        valueForNode(inputs[i]), offsets);
+      // Look for a series of the same Node being concated consecutively many
+      // times. We can wrap n such consecutive repeats into a single insert with
+      // count n along the dim axis.
+      for (size_t i = 0, j = 1, e = inputs.size(); i < e; i = j++) {
```

Comment:
Could you extract this into a method that checks:  bool isConsecutiveSomething(...) ?

Reply:
Done -- I moved to a `getConsecutiveSameNodeCount()` -- I think it makes more sense than an `isConsecutive()` function.